src/buffer.rs

```diff
@@ -921,6 +921,35 @@ where
         self.get_pixel_mut(x, y).blend(&p)
     }
 
+    fn copy_within(&mut self, from: (u32, u32), to: (u32, u32), width: u32, height: u32) -> bool {
+        let (fx, fy) = from;
+        let (tx, ty) = to;
+        if tx.max(fx) + width > self.width() || ty.max(fy) + height > self.height() {
```

Comment:
This is also prone to overflow, consider `tx = 2`, `width = u32::MAX`. I don't have a problem with panicking on such input, `std::slice::copy_within` also does this, but it should happen more predictably. For example, first assert that both coordinates lie within the boundaries. Then the calculation `self.width() - tx.max(fx)` is safe and can be compared to `width` as an alternative check.

Reply:
Ah, I completely forgot about overflow becoming a problem, addressed the points you mentioned.